analyzers/tests/SonarAnalyzer.UnitTest/TestCases/CloudNative/AzureFunctionsLogFailures.cs

```diff
@@ -0,0 +1,325 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+public static class LogInCatchClause
+{
+    [FunctionName("Sample")]
+    public static void EmptyCatchClause(ILogger log)
+    {
+        try { }
+        catch { } // Noncompliant {{Log exception via ILogger with LogLevel Information, Warning, Error, or Critical.}}
+//      ^^^^^
+    }
+
+    [FunctionName("Sample")]
+    public static void LogLevelInvalid(ILogger log)
+    {
+        try { }
+        catch // Noncompliant
+//      ^^^^^
+        {
+            log.LogTrace(string.Empty);        // Secondary
+//          ^^^^^^^^^^^^^^^^^^^^^^^^^^
+            log.LogDebug(string.Empty);        // Secondary
+        }
+    }
+
+    [FunctionName("Sample")]
+    public static void LogExceptionInCatchClause(ILogger log)
+    {
+        try { }
+        catch (Exception ex) // Compliant
+        {
+            log.LogError(ex, "");
+        }
+    }
+
+    [FunctionName("Sample")]
+    public static void LogExceptionInWrappedLogger(ILogger log)
+    {
+        try { }
+        catch (Exception ex) // Compliant
+        {
+            var nullLogger = NullLogger.Instance;
+            nullLogger.Log(LogLevel.Error, new EventId(), (object)null, ex, (s, e) => string.Empty);
+        }
+    }
+
+    [FunctionName("Sample")]
+    public static void LogExceptionLoggedInLocalFunction(ILogger log)
+    {
+        try { }
+        catch (Exception ex) // FN. Any call to "log" is considered valid. Reachability is not considered.
+        {
+            void Log() => log.LogError("");
+        }
+    }
+
+    [FunctionName("Sample")]
+    public static void LogExceptionLoggedInLambda(ILogger log)
+    {
+        try { }
+        catch (Exception ex) // FN. Any call to "log" is considered valid. Reachability is not considered.
+        {
+            Action x = () => log.LogError("");
+        }
+    }
+
+    [FunctionName("Sample")]
+    public static void LogExceptionLoggedInAnonymousFunction(ILogger log)
+    {
+        try { }
+        catch (Exception ex) // FN. Any call to "log" is considered valid. Reachability is not considered.
+        {
+            Action x = delegate () { log.LogError(""); };
+        }
+    }
+
+    [FunctionName("Sample")]
+    public static void LogExceptionIsPassedInLambda(ILogger log)
+    {
+        try { }
+        catch (Exception ex) // FN. Any call to "log" is considered valid. Reachability is not considered.
+        {
+            DoSomething(() => log.LogError(""));
+        }
+
+        void DoSomething(Action a) => a();
```

Comment:
Because it is not a FN (for a wrong reason) now :) Let's make it authentic.
```suggestion
        void DoSomething(Action a) { };
```